testcases/OpTestGSBStaticKey.py

```diff
@@ -128,86 +120,59 @@ def getPRePDisk(self):
         Parameter enable=True for enabling Secure Boot
         Parameter enable=False for disabling Secure Boot
         '''        
-        self.checkKernel()
-        if self.distro == 'rhel' and enable:
-            # Get the PReP disk file
-            self.getPRePDisk()
-            # Get the all the required files
-            self.getRHELFiles()
-            self.checkGrub()
-            # Running grub2-install on PReP disk
-            out = self.c.run_command("grub2-install %s" % self.prepDisk)
-            if "Installation finished. No error reported." not in out:
-                self.fail("RHEL: Failed to install on PReP partition")
-            # Copy the signed grub in to the PReP disk using 'dd' command
-            out = self.c.run_command("dd if=%s of=%s || echo 'no'" %
-                                     (self.grubFilename, self.prepDisk))
-            if "no" in "".join(out):
-                self.fail("RHEL: Failed to copy to PReP partition")
-        elif self.distro == 'rhel' and not enable:
-            # Nothing to do for Secure Boot disable for RHEL at OS level
-            pass
-        bfile = "/etc/sysconfig/bootloader"
-        if self.distro == 'sles' and enable:
-            # Add SECURE_BOOT="yes" at file  /etc/sysconfig/bootloader
-            cmd = "sed -i '/SECURE_BOOT=\"no\"/c\SECURE_BOOT=\"yes\"' %s" % bfile
-            self.c.run_command(cmd)
-            out = self.c.run_command("pbl --install || echo 'no' ")
-            if "no" in "".join(out):
-                self.fail("SLES: Failed to enable Secure Boot")
-        elif self.distro == 'sles' and not enable:
-            # Add SECURE_BOOT="no" at file  /etc/sysconfig/bootloader
-            cmd = "sed -i '/SECURE_BOOT=\"yes\"/c\SECURE_BOOT=\"no\"' %s" % bfile
-            self.c.run_command(cmd)
-            out = self.c.run_command("pbl --install || echo 'no'")
-            if "no" in "".join(out):
-                self.fail("SLES: Failed to disable Secure Boot")
+        if self.kernel_signature == "signed_kernel":
+            if self.distro == 'rhel' and enable:
+                # Get the PReP disk file
+                self.getPRePDisk()
+                #Proceed ahead only if the grub is signed
+                if self.grub_signature == "signed_grub":
+                    # Running grub2-install on PReP disk
+                    out = self.c.run_command("grub2-install %s" % self.prepDisk)
+                    if "Installation finished. No error reported." not in out:
+                        self.fail("RHEL: Failed to install on PReP partition")
+                    # Copy the signed grub in to the PReP disk using 'dd' command
+                    out = self.c.run_command("dd if=%s of=%s || echo 'no'" % 
+                                             (self.grub_filename, self.prepDisk))
+                    if "no" in "".join(out):
+                        self.fail("RHEL: Failed to copy to PReP partition")
```

Comment:
Same here. In case of a failure, should we restore the grub2 to original state. If we simply fail the testcase, will reboot of system leave it in un-bootable state?

Reply:
yes good idea Sachin ,

Reply:
Not tested in this way.

Well, to test it - how can this scenario be reproduced?
1] Introducing a failure in the PreP copying.
2] Then in case of a failure - restoring the grub2 to the original state - How is it done?

Reply:
Done